stock_barcodes/wizard/stock_production_lot.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2019 Sergio Teruel <sergio.teruel@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import fields, models
+
+
+class WizStockBarcodesNewLot(models.TransientModel):
+    _inherit = "barcodes.barcode_events_mixin"
+    _name = "wiz.stock.barcodes.new.lot"
+    _description = "Wizard to create new lot from barcode scanner"
+
+    product_id = fields.Many2one(comodel_name="product.product", required=True)
+    lot_name = fields.Char(string="Lot name", required=True)
+
+    def on_barcode_scanned(self, barcode):
+        product = self.env["product.product"].search([("barcode", "=", barcode)])[:1]
+        if product and not self.product_id:
+            self.product_id = product
+            return
+        self.lot_name = barcode
+
+    def _prepare_lot_values(self):
+        return {
+            "product_id": self.product_id.id,
+            "name": self.lot_name,
+            "company_id": self.env.user.company_id.id,
```

Comment:
Getting company from the user causes problems since this is the default company, not the current company. Perhaps it should be defined here and default is passed from calling document.

Reply:
Yes, good catch. It should be `self.env.company.id` AFAIK. Can you do the PR?

Reply:
Or maybe the company from the picking if locked.

Reply:
I am busy right now working on mrp_barcodes module :)

Reply:
@sergio-teruel can you please check?

Reply:
@levkar eagering to see mrp_barcodes :wink: